gammapy/stats/feldman_cousins.py

```diff
@@ -0,0 +1,436 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Provide a Feldman Cousins algorithm."""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import numpy as np
+import matplotlib.pylab as plt
```

Comment:
matplotlib and scipy are optional dependencies for Gammapy, i.e. having module-level imports are not OK (will make the test fail on travis-ci -> can't be merged).

The pattern for optional dependencies is to put the import statements only in the functions / methods where they are used. (Yes, this can be repetitive and is not very pretty. But efficiency is OK ... Python never re-loads modules, import just looks them up from a dict.)

Then in tests you have to do this little dance:
http://astropy.readthedocs.org/en/latest/development/testguide.html#tests-requiring-optional-dependencies

This is what you should do for scipy.

Concerning matplotlib: the only place where you use it is the `plot` option in `get_upper_and_lower_limit`.
Usually it's not good to mix computation and plotting code, so I'd be +1 to just remove that option and show an example in the docs how to plot this.
If you think  this is something users commonly want, a separate `plot_...` function is better than a `plot` option in the computation method.